pa4/checker-pa4.ml

```diff
@@ -481,12 +512,13 @@ let rec exp_typecheck(o_e: obj_env) (m_e: method_env) (exp: expression) : static
 					method_identifier_line_number method_name caller_type_name ;
 			exit 1
 		end
-	end
-	| New(class_identifier) -> begin
-		printf "Doing a New\n" ;
-		let (_, class_name) = class_identifier in
-		Class(class_name)
-	end
+	end	
+	| New(e1) -> 
+		let (id_location, id_name) = e1 in
+		(match id_name with
+			| "SELF_TYPE" -> SELF_TYPE(type_to_str c_e)
```

Comment:
How this the `type_to_str` here work for `c_e`? Shouldn't there be a problem with it?

Reply:
I think thats how it should be done. We could confirm with Cuong. @thanvietcuong

Reply:
So is `c_e` a `Class`? I thought that it would be a Hash Table like the other environments.